scripts/launch_workflow_cf/utils.py

```diff
@@ -5,94 +5,168 @@
+      workspace_name=workspace_name,
+      method_namespace=method_namespace,
+      method_name=method_name,
+      entity=entity_set_name,
+      etype=set_entity_type,
+      expression=expression,
+      headers=headers
+  )
+  print("Submission response:\n", json.dumps(create_submisson_response.json(), indent=4))
+  check_fapi_response(response=create_submisson_response, success_code=201)
+  submission_id = create_submisson_response.json()["submissionId"]
+  print(f"Successfully created submission: submissionId = {submission_id}.")
+
+def get_access_token(secret_path):
+  """Conditionally create an access token with the minimum necessary scopes.
+
+  When run as a Cloud Function, a service account's JSON credentials file from Secret Manager
+  is used to populate the access token. When run locally, either the service account JSON key file or the
+  application default credential is used to populate the access token.
+
+  Args:
+    secret_path: The 'Resource ID' of the service account key stored in Secret Manager. Or, if
+      testing locally, the filepath to the JSON key for the service account.
+  Returns:
+    An access token.
+  """
+  scopes = ["https://www.googleapis.com/auth/userinfo.profile", "https://www.googleapis.com/auth/userinfo.email"]
+
+  if not secret_path:  # Running locally as a user.
+    credentials = GoogleCredentials.get_application_default()
+    credentials = credentials.create_scoped(scopes)
+  elif os.path.isfile(secret_path):  # Running locally as the service account.
+    credentials = ServiceAccountCredentials.from_json_keyfile_name(secret_path, scopes=scopes)
+  else: # Running inside the Cloud Function.
+    # Retrieve the secret from the secret manager API.
+    client = SecretManagerServiceClient()
+    response = client.access_secret_version(secret_path)
+    service_account_key = response.payload.data.decode("utf-8")
+    json_acct_info = json.loads(service_account_key)
+    credentials = ServiceAccountCredentials.from_json_keyfile_dict(json_acct_info, scopes=scopes)
+
+  return credentials.get_access_token().access_token
+
+# Helper methods for Firecloud API calls. We cannot use the methods in firecloud.api because
+# we need to explicitly pass a header for auth.
 
 def check_fapi_response(response, success_code):
-    if response.status_code != success_code:
-        print(response.content)
-        raise ferrors.FireCloudServerError(response.status_code, response.content)
-
-### Firecloud API calls
+  if response.status_code != success_code:
+    print(response.content)
+    raise FireCloudServerError(response.status_code, response.content)
 
-def get_workspace_config(namespace, workspace, cnamespace, config, headers):
-    uri = f"https://api.firecloud.org/api/workspaces/{namespace}/{workspace}/method_configs/{cnamespace}/{config}"
-    return requests.get(uri, headers=headers)
+def get_workflow_method_config(workspace_namespace, workspace_name, method_namespace, method_name, headers):
```

Comment:
my goodness, thank you for catching and fixing the completely inaccurate function names!